packages/frontend/src/features/sqlRunner/components/Header/HeaderEdit.tsx

```diff
@@ -113,9 +146,9 @@ export const HeaderEdit: FC = () => {
 
                     <Group spacing="md">
                         <Button
-                            variant="default"
                             size="xs"
-                            disabled={!config || !sql}
+                            color={'green.7'}
+                            disabled={!config || !sql || !hasChanges()}
                             loading={isLoading}
                             onClick={() => {
```

Comment:
I initially tried that, but it was not saving the right config on save, I believe is because a change that happens on the mutation (or something else) . The only way to fix it was using the `data + reset` combo

Reply:
See bug, I need to press save twice to correctly load the sql chart state

[Screencast from 09-09-24 16:06:02.webm](https://github.com/user-attachments/assets/d7cfcc7a-0c8c-4575-ad3a-a993ca9ceaba)

Reply:
Ok, cool, I'm absolutely ok with keeping the function _inline_ so all good :)

Reply:
Fixed it using callback in the end